### PR TITLE
Display an error when a user is not allowed to accept or reject a feed invitation

### PR DESCRIPTION
**Pull request by caiosba**
## Description

Display an error page when a user is not allowed to accept or reject a feed invitation.

- Adding (optional) props `title` and `description` to `<NotFound />` component
- Adding a couple of unit tests to the `<NotFound />` component
- Using the `<NotFound />` component with custom title and description when a user clicks on a feed invitation but they are not an admin of that workspace

Fixes: CV2-4122.

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

I added unit tests for the `<NotFound />` component, but the fix itself can be tested manually: invite to a feed a user that is a member of at least two teams: A and B. In team A, that user is an admin, while in team B, that user is an editor.

- When accessing the feed invitation from team A page, the user should be able to see the invitation with "accept" and "reject" buttons.
- When accessing the feed invitation from team B page, the user should see an error page with custom title and description.

## Things to pay attention to during code review

The permission is `create FeedTeam` because it's actually what happens when an invitation is accepted. For simplicity sake, I didn't check the permission for rejection, since it's the same logic.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [x] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

### CODE REVIEW

**Comment by brianfleming** [comment]
@caiosba its not working for me. I am going to reset the data locally to see if its an old data issue, but I invited another user who is an editor to a feed and I am still able to get to the shared feed invite:

### Here I am `ali@example.org`, I am an editor of workspace `hudson-corwin-and-ritchie`
Screenshot 2023-12-22 at 9 39 53 AM

### Invite from `elwood@example.org`
Screenshot 2023-12-22 at 9 39 36 AM

**Comment by caiosba** [comment]
Hum... @brianfleming, just to confirm, ali@example.com is not a super-admin, right? And also please be sure you're using the latest version of Check API.

**Comment by brianfleming** [comment]
> Hum... @brianfleming, just to confirm, [ali@example.com](mailto:ali@example.com) is not a super-admin, right? And also please be sure you're using the latest version of Check API.

I will make sure I am latest on API (that may be the issue)
I don't think ali@example.org is a super admin, thats an account created via the seed script, which I don't believe creates super admins...right?

**Comment by caiosba** [comment]
> > Hum... @brianfleming, just to confirm, [ali@example.com](mailto:ali@example.com) is not a super-admin, right? And also please be sure you're using the latest version of Check API.
> 
> I will make sure I am latest on API (that may be the issue) I don't think [ali@example.org](mailto:ali@example.org) is a super admin, thats an account created via the seed script, which I don't believe creates super admins...right?

Ohh looks like the users are super-admins:

- https://github.com/meedan/check-api/blob/develop/db/seeds.rb#L366
- https://github.com/meedan/check-api/blob/develop/db/seeds.rb#L275

We need to remove the `is_admin` attributes from the seed script (cc @vasconsaurus).

@brianfleming you confirm / fix it this way:

```
$ docker-compose exec api bundle exec rails c
u = User.find_by_email('ali@example.com')
puts u.is_admin # Should print "true" or "false"
u.is_admin = false
u.save!
```

**Comment by brianfleming** [comment]
> puts u.is_admin

yep, you are right, user was `is_admin`
I'm going to reset all my data and take a look again

**Comment by vasconsaurus** [comment]
> We need to remove the is_admin attributes from the seed script (cc @vasconsaurus).

@caiosba I will open a ticket to fix this

**Review by brianfleming — APPROVED** [review]
Ok, I retested this @caiosba after update the admin thing and resetting my data and I think everything looks good now.

You will see I just opened a PR against this branch to update the styles to CDS styles. I think we should merge that into your PR:

[1784](https://github.com/meedan/check-web/pull/1784)

**Comment by caiosba** [comment]
Done @brianfleming thanks - @dariusk , I appreaciate a last review more from the code side, please :)

**Comment by brianfleming** [comment]
> Done @brianfleming thanks - @dariusk , I appreaciate a last review more from the code side, please :)

teamwork

File: src/app/components/NotFound.js [hunk 1]
```diff
@@ -18,44 +19,62 @@ const NotFound = () => (
...
   </PageTitle>
 );
 
+NotFound.defaultProps = {
+  title: null,
+  description: null,
+};
+
+NotFound.propTypes = {
+  title: PropTypes.node, // <FormattedMessage />, <FormattedHTMLMessage />, <Element>String</Element>, etc.
```
Comment (dariusk) [inline src/app/components/NotFound.js:76]: Glad to see the comments here, very helpful

File: src/app/components/error/ErrorPage.js [hunk 2]
```diff
@@ -12,16 +13,22 @@ const ErrorPage = ({
...
+          />
+        )}
+      </FormattedGlobalMessage>
+      <h6>{cardTitle}</h6>
+      <p>
+        {cardText}
+      </p>
+    </div>
   </PageTitle>
 );
```
Comment (dariusk) [inline src/app/components/error/ErrorPage.js:34]: We need `PropTypes` defined for this component

**Review by dariusk — CHANGES_REQUESTED** [review]
Looks good, we can just add those PropTypes I pointed out and it's good to go.

**Comment by caiosba** [comment]
> Looks good, we can just add those PropTypes I pointed out and it's good to go.

Done, thanks!

**Comment by brianfleming** [comment]
my bad on that lint failure @caiosba , I didn't fully deprecate some code in the other PR

**Comment by caiosba** [comment]
> my bad on that lint failure @caiosba , I didn't fully deprecate some code in the other PR

Always good to see `FIXME`'s getting actually fixed :D

**Comment by caiosba** [comment]
@dariusk can you please re-review? I implemented the missing `PropTypes` and fixed the unit tests.

**Review by dariusk — APPROVED** [review]

**Pull request merged**